Feature/admin dashboard

Istott opened the pull request:
Created dashboard component with GET request endpoint with mock data. Added search bar and drop down option selector for orders. Added less file for dashboard index styling.

heraimenka committed:
feat: read mock order data in admin dashboard
Co-authored-by: Isaac Stott <isaac.cloyd@gmail.com>

heraimenka committed:
feat: add header links

heraimenka committed:
feat: move logout link to header, edit styles
Co-authored-by: Isaac Stott <isaac.cloyd@gmail.com>
Co-authored-by: William Sedlacek <wsedlacekc@gmail.com>

heraimenka committed:
feat: edit styles for orders on dashboard

Istott committed:
feat: add search bar and order select options.

add less file for dashboard index

Co-authored-by: Alexander Heraimenka <a.heraimenka@gmail.com>

wSedlacek commented on CODE_OF_CONDUCT.md:
```diff
@@ -14,22 +14,22 @@ appearance, race, religion, or sexual identity and orientation.
 
 Examples of unacceptable behavior by participants include:
 
--   The use of sexualized language or imagery and unwelcome sexual attention or
-    advances
--   Trolling, insulting/derogatory comments, and personal or political attacks
--   Public or private harassment
--   Publishing others' private information, such as a physical or electronic
-    address, without explicit permission
--   Other conduct which could reasonably be considered inappropriate in a
-    professional setting
+- The use of sexualized language or imagery and unwelcome sexual attention or
+  advances
+- Trolling, insulting/derogatory comments, and personal or political attacks
+- Public or private harassment
+- Publishing others' private information, such as a physical or electronic
+  address, without explicit permission
+- Other conduct which could reasonably be considered inappropriate in a
+  professional setting
```
Let's revert this file and make a separate PR to format any files

heraimenka replied:
Changed it to its original version

wSedlacek commented on src/components/common/Header.js:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+export default function Header() {
+  return (
+    <header>
+      <ul>
+        <li>
+          <Link to="/">Home</Link>
+        </li>
+        <li>
+          <Link to="/dashboard">Dashboard</Link>
+        </li>
+      </ul>
+    </header>
+  );
+}
```
This doesn't appear to be used anywhere?
Is this for future development or is this simply zombie code?

heraimenka replied:
Zombie! Deleted the file

wSedlacek commented on src/components/common/Order.js:
```diff
@@ -0,0 +1,43 @@
+  const { order, formattedDate } = props;
+
+  return (
+    <>
```
Should this be a `<div>` or some other element rather then a fragment so that these are grouped together on the DOM?
Would we need to apply a class or id etc here in the future?

heraimenka replied:
Changed to div

wSedlacek commented on src/components/common/Order.js:
```diff
@@ -0,0 +1,43 @@
+
+  return (
+    <>
+      <h4>Status:</h4>
```
I noticed that we have `<h4>`, but the place where this is used previously uses an `<h1>` so what happened to `<h2>` and `<h3>`? 

Remember [Avoid skipping heading levels: always start from h1, next use h2 and so on.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#Usage_notes)

heraimenka replied:
Changed h4 to h2

wSedlacek commented on src/components/common/Order.js:
```diff
@@ -0,0 +1,43 @@
+      </p>
+      {order.comments && (
+        <>
+          <h4>Comments:</h4>
```
Should this header be above all the comments instead of for each comment?

wSedlacek commented on src/components/common/TopHeader.js:
```diff
@@ -0,0 +1,35 @@
+  const { Header } = Layout;
+
+  function logout(event) {
+    event.preventDefault();
```
Is `preventDefault()` necessary here?

heraimenka replied:
Removed the line with preventDefault

wSedlacek commented on src/components/pages/Dashboard/dashboard.less:
```diff
@@ -0,0 +1,12 @@
+    align-items: baseline;
+    justify-content: space-between;
+
+    .searchBar{
+        border: none;
+        border-bottom: 1px black solid;
+        background-color: transparent;
+    }
```
Is there a reason `.searchBar` should be nested? Does the fact that is is in `.dashboard` affect the reason as to why these styles would be applied?

heraimenka replied:
Moved .searchBar to the same level

hera replied:
Test

wSedlacek commented on src/components/pages/Dashboard/index.js:
```diff
@@ -0,0 +1,65 @@
+  // load orders on first render
+  useEffect(props.ordersGet, []);
+
+  const { Title, Paragraph } = Typography;
+  const { Panel } = Collapse;
```
What was happening when this was outside of the component ie on line 10 or above?
Did we have errors?

wSedlacek reviewed (changes requested):
We are lacking some test that will be required to merge and one test has broken. Please fix these test before merging.

I also had a few questions on the code. Please reply to these before requesting rereview.

heraimenka committed:
revert: change CODE_OF_CONDUCT.md to its original version

heraimenka committed:
refactor: minor changes for a pull request

Commit:
revert: remove formatted files from PR

These changes aren't necessary for this PR but should be included in a future PR focused on formatting

wSedlacek commented:
We also do need the test to merge this PR. So make sure to get those done.

heraimenka committed:
refactor: move antd destr. outside components

wSedlacek commented on src/components/common/TopHeader.js:
```diff
@@ -0,0 +1,33 @@
+
+export default function TopHeader() {
+  const { authService, authState } = useOktaAuth();
+  const { Header } = Layout;
```
Can we also move this out of the component?

heraimenka committed:
refactor: move antd destr. outside the component

heraimenka committed:
refactor: modify code to have the test passing

wSedlacek reviewed (approved):
Awesome work Alex and Issac!
